tests/nstack.rs

```diff
@@ -106,66 +100,25 @@ fn branch_lengths() {
     }
 }
 
-#[derive(Clone, Debug, Archive)]
-#[archive(as = "Self")]
-#[archive(bound(archive = "
-  K: Primitive,
-  MaxKey<K>: Primitive,
-"))]
-struct MaxAndCardinality<K> {
-    cardinality: Cardinality,
-    max: MaxKey<K>,
-}
-
-impl<K> Default for MaxAndCardinality<K> {
-    fn default() -> Self {
-        Self {
-            max: Default::default(),
-            cardinality: Default::default(),
-        }
-    }
-}
+#[test]
+fn persist() {
+    let n: u64 = 1024;
 
-impl<K> Borrow<Cardinality> for MaxAndCardinality<K> {
-    fn borrow(&self) -> &Cardinality {
-        &self.cardinality
+    let mut stack = NStack::<_, Cardinality>::new();
+    for i in 0..n {
+        stack.push(i);
     }
-}
 
-impl<K> Borrow<MaxKey<K>> for MaxAndCardinality<K> {
-    fn borrow(&self) -> &MaxKey<K> {
-        &self.max
-    }
-}
+    let stored = Portal::put(&stack);
+    let restored = Portal::get(stored);
 
-impl<K, L> Annotation<L> for MaxAndCardinality<K>
-where
-    L: Keyed<K>,
-    K: Primitive + Clone + Ord,
-{
-    fn from_leaf(leaf: &L) -> Self {
-        Self {
-            cardinality: Cardinality::from_leaf(leaf),
-            max: MaxKey::from_leaf(leaf),
-        }
+    // empty original
+    for i in 0..n {
+        assert_eq!(stack.pop().unwrap(), n - i - 1);
     }
-}
 
-impl<A, K> Combine<A> for MaxAndCardinality<K>
-where
-    A: Borrow<Cardinality> + Borrow<MaxKey<K>>,
-    K: Clone + Ord,
-{
-    fn combine<C>(anno: AnnoIter<C, A>) -> Self
-    where
-        C: Compound<A> + Archive,
-        C::Archived: ArchivedCompound<C, A>,
-        C::Leaf: Archive,
-        A: Annotation<C::Leaf>,
-    {
-        Self {
-            cardinality: Cardinality::combine(anno.clone()),
-            max: MaxKey::combine(anno),
-        }
+    // empty restored copy
```

Comment:
minor nit, this is incorrect, should say something like 'check restored copy'

Reply:
You're right, it's not emptying anything. Fixed